[Gutenberg] Hide help button from UBE

SiobhyB opened the pull request:
Addresses https://github.com/wordpress-mobile/gutenberg-mobile/issues/4339

## Related PRs and Installable Builds

* `Gutenberg`: https://github.com/WordPress/gutenberg/pull/37221
* `Gutenberg Mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4352
* `WPiOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/17644

## Description

As described in https://github.com/wordpress-mobile/gutenberg-mobile/issues/4339, a help button is currently displaying in the bottom right-hand corner of the unsupported block editor (UBE) in both the Android and iOS apps. The button can be seen while logged into WordPress.com. When tapped on, the button prompts users to navigate away from the editor on Android, creating the potential for edits to be lost.

With this PR, the button is hidden from view in order to avoid confusion and potential loss of edits. A high-level overview of the code changes can be found in the `Types of changes` section of [the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/37221).

## Testing

Please refer to [the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/37221) as the "central PR" with the most up-to-date testing instructions. 

<hr />

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

SiobhyB committed:
Update reference to Mobile Gutenberg

SiobhyB committed:
Fetch external styles from Gutenberg Mobile

'getOnGutenbergReadyExternalStyles' will fetch and return external styles defined in Gutenberg Mobile's 'external-style-overrides.css' file.

SiobhyB committed:
Merge branch 'develop' into gutenberg/update/hide-help-button-from-ube

SiobhyB committed:
Merge branch 'develop' into gutenberg/update/hide-help-button-from-ube

twstokes commented on libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/WPGutenbergWebViewActivity.java:
```diff
@@ -138,6 +138,11 @@ private String getAuthenticationPostData(String authenticationUrl,
         return "";
     }
 
+    protected String getOnGutenbergReadyExternalStyles() {
```
@SiobhyB do you think we should add `@Override` to this method? I believe it's optional, but will throw a compiler warning if the `GutenbergWebViewActivity` superclass changes its method name in the future and we forget to update it.

SiobhyB replied:
That makes sense, thanks for spotting that @twstokes! I've gone ahead to add it in https://github.com/wordpress-mobile/WordPress-Android/pull/15674/commits/ce5c938a09ff5d9e721ec6265ae864bbad23d5f0.

SiobhyB committed:
Add @override to avoid compiler error

Added following feedback here: https://github.com/wordpress-mobile/WordPress-Android/pull/15674#discussion_r777798174

twstokes reviewed (approved):
LGTM @SiobhyB to merge after pointing to the appropriate tag. 🎉

SiobhyB committed:
Merge branch 'develop' into gutenberg/update/hide-help-button-from-ube

SiobhyB committed:
Update reference to Mobile Gutenberg
